store/storethehash/storethehash.go

```diff
@@ -290,7 +291,10 @@ func (it *sthIterator) Next() (multihash.Multihash, []indexer.Value, error) {
 			continue
 		}
 
-		origMultihash := multihash.Multihash(dm.Digest[:len(dm.Digest)-len(indexKeySuffix)])
+		var code uint64
+		code = binary.BigEndian.Uint64(dm.Digest[len(dm.Digest)-binary.Size(code)-len(indexKeySuffix) : len(dm.Digest)-len(indexKeySuffix)])
+		origDigest := dm.Digest[:len(dm.Digest)-len(indexKeySuffix)-binary.Size(code)]
+		origMultihash, _ := multihash.Encode(origDigest, code)
```

Comment:
I think this also needs to be done in `RemoveProvider` [here](https://github.com/filecoin-project/go-indexer-core/pull/38/files#diff-b0444f44707cb9e7710ecf06ea6bfeea3879a6e115be960097dc85070db65e62L139)